pkg/policy/api/l4.go

```diff
@@ -55,6 +55,59 @@ func (p PortProtocol) Covers(other PortProtocol) bool {
 	return true
 }
 
+// Secret is a reference to a secret, backed by k8s or local file system.
+type Secret struct {
+	// Namespace is the namespace in which the secret exists.
+	// Context of use determines the default value if left out (e.g., "default")
+	//
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// Name is the name of the secret.
+	Name string `json:"name"`
+}
+
+// Equal returns true if 'a' and 'b' have the same contents.
+func (a *Secret) Equal(b *Secret) bool {
+	return a == nil && b == nil || a != nil && b != nil && *a == *b
```

Comment:
Why don't you have the receiver and the argument to be non pointers?

Reply:
This has been deleted in a later commit in the series.